README.md

```diff
@@ -1,13 +1,132 @@
-# HEPML Installer
+# HEPML Environment
 
-Installs a minimal Python3 environment for HEP machine learning
+Provides a minimal Python3 environment for machine learning focused towards HEP use cases
 
 [![License](https://img.shields.io/badge/License-BSD%203--Clause-blue.svg)](https://opensource.org/licenses/BSD-3-Clause)
 [![Build Status](https://travis-ci.com/matthewfeickert/HEPML-env.svg?branch=master)](https://travis-ci.com/matthewfeickert/HEPML-env)
 
+---
+
+Setting up a machine learning environment has gotten easier recently, but there are at times still problems that arise from time to time. HEPML-env allows for easily setting up a standard machine learning Python environment that should allow you to get to work with HEP data immediately. It should be machine agnostic, such that it can setup an identical environment on your laptop or on LXPLUS.
+
+### The default environment
+
+- [numpy](https://github.com/numpy/numpy)
+- [scipy](https://github.com/scipy/scipy)
+- [matplotlib](https://github.com/matplotlib/matplotlib)
+- [pandas](https://github.com/pandas-dev/pandas)
+- [h5py](https://github.com/h5py/h5py)
+- [uproot](https://github.com/scikit-hep/uproot)
+- [scikit-learn](https://github.com/scikit-learn/scikit-learn)
+- [scikit-image](https://github.com/scikit-image/scikit-image)
+- [tensorflow](https://github.com/tensorflow/tensorflow)
+- [jupyter](https://github.com/jupyter)
+
+
+### Core features
+- Reproducibility
+- Portability
+- ROOT-less
+
+## Requirements
+
+- GNU
+    - gcc/g++ (5.4 or higher is recommended)
+    - zlibc
+    - zlib1g-dev
+    - libssl-dev
+    - wget
+    - make
+    - Bash
+- Python 3.6 or higher
+- [pipenv](https://docs.pipenv.org/)
+
+HEPML uses Python 3 and pipenv, but if you don't have them installed already the [`install.sh`](https://github.com/matthewfeickert/HEPML-env/blob/master/scripts/install.sh) will take care of that for you.
+
+<details>
+ <summary>Installing GNU requirements on Debian/Ubuntu</summary>
+
+On Debian/Ubuntu the GNU requirements can be met by
+
+```
+apt install gcc g++ git zlibc zlib1g-dev libssl-dev wget make
+```
+
+</details>
+
+## Installation and setup
+
+1. Clone the repo
+
+<details>
+ <summary>Note on cloning on LXPLUS</summary>
+
+When trying to use SSH with GitHub on LXPLUS it is important to make sure that your `~/.ssh/config` is properly configured. It may need to contain something along the lines of
+
+```
+Host github.com
```

Comment:
This is a useful tip: I only recently realized that you can use different identity files for different sites. I suppose the reason this is better than what I do (upload the same `id_rsa.pub` key everywhere) is that I don't have to put my private key for other things on every cluster where I just need github access?

Reply:
Yeah, and then also if you need to regenerate your key for whatever reason you only need to worry about moving that new key around while your other keys are unaffected, instead of having it affect _everything_ instantly.